### ISSUE TITLE
form-data file is empty with node version 14.1.0

### ISSUE THREAD

**Issue by chwo** [pillarjs/multiparty#226]
## Description

I am doing a `multipart/form-data` POST Request with fields and a file in the body.

When I try to access the file in the `form.parse` function, the `files` property is empty. 

```typescript
const form = new multiparty.Form();
form.parse(req, (error, fields, files) => {
    console.log(files) // -> is empty: {}
});
```

I tried debugging it and noticed that 

https://github.com/pillarjs/multiparty/blob/f4c2ea662e0e5e9b5bb58d6a729c1ee5bb2b2ab2/index.js#L134

is called before

https://github.com/pillarjs/multiparty/blob/f4c2ea662e0e5e9b5bb58d6a729c1ee5bb2b2ab2/index.js#L130

This order occurs with node version `14.1.0` whereas for example with node version `12.0.0` the order is the opposite and the `files` property contains the file.

## Environment
- multiparty version `4.2.1`
- node version `14.1.0`

**Comment by dougwilson** [comment]
Hi @chwo thank you for the report. It is indeed a bug due to some streams changes in Node.js 14. I am working to get it all fixed up. You can look though https://github.com/expressjs/connect-multiparty/issues/29 if you wanted to know about some of the background, but otherwise I'll use this issue as the tracker for the Node.js 14 fix in multiparty itself 👍

**Comment by hkumar4** [comment]
Hi, Can someone please update when this issue would be fixed? I see that the form.on('close') fires twice since node 14. Once with empty files and then with files populated. Thank you for looking into this.

**Comment by barisusakli** [comment]
Hi @dougwilson, looks like you fixed this, any eta on a new release for this and connect-multiparty?

**Comment by dougwilson** [comment]
Hi @barisusakli indeed, I fixed both issues with the Node.js changes: max size limit broken and files empty. The first change was an issue with a dependency being broken and it looks like it will just never be fixed, so that was a bit more involved change to test. I think it is working well, but if not we'll just field any reported bugs. I'm going to land them here now, sorry for the delay.

**Comment by dougwilson** [comment]
The fix is merged into `master`. Please feel free to test ahead of time. I try not to release and then go to sleep, so will release first thing in the morning.

**Comment by barisusakli** [comment]
@dougwilson from my testing our upload tests pass by using the latest code from multiparty with node 14.6.0 and connect-multiparty 2.2.0

**Comment by dougwilson** [comment]
Published as patch version 4.2.2

**Comment by barisusakli** [comment]
👍 waiting on connect-multiparty

**Comment by dougwilson** [comment]
@barisusakli it is picked up automatically as the `package.json` is `~4.2.1`
